bundles/org.eclipse.e4.ui.workbench.renderers.swt/src/org/eclipse/e4/ui/workbench/renderers/swt/StackRenderer.java

```diff
@@ -623,6 +667,38 @@ public Object createWidget(MUIElement element, Object parent) {
 		return tabFolder;
 	}
 
+	/**
+	 * @param tabFolder
+	 */
+	private void createOnboardingControls(OnboardingCTabFolder tabFolder) {
+		Composite container = WidgetFactory.composite(SWT.None).layout(GridLayoutFactory.swtDefaults().create())
+				.create(tabFolder);
+
+		GridDataFactory gridData = GridDataFactory
+				.create(GridData.VERTICAL_ALIGN_CENTER | GridData.HORIZONTAL_ALIGN_CENTER).grab(true, true);
+
+		Composite content = WidgetFactory.composite(SWT.NONE).supplyLayoutData(gridData::create)
+				.layout(GridLayoutFactory.swtDefaults().create()).create(container);
+
+		Color color = container.getDisplay().getSystemColor(SWT.COLOR_WIDGET_DISABLED_FOREGROUND);
+		welcomeText = WidgetFactory.label(SWT.NONE).foreground(color).supplyLayoutData(gridData::create)
+				.create(content);
+
+		welcomeImage = WidgetFactory.label(SWT.NONE).foreground(color).supplyLayoutData(gridData::create)
+				.create(content);
+
+		tabFolder.setOnboarding(container);
+
+		tabFolder.addDisposeListener(e -> {
+			if (!color.isDisposed()) {
+				color.dispose();
```

Comment:
Colors don't need to be disposed anymore.

Reply:
removed it